File: 20200402/code.js
function solution(arr, k) {
  count = arr.reverse().reduce((acc, cur) => {
    if (parseInt(k / cur) > 0) {
      acc += parseInt(k / cur);
      k %= cur;
    }
    return acc 
  }, 0)

  return count;

}


console.log(solution([1,5,10,50,100,500,1000,5000], 4790))

File: 20200402/code.py
def solution():
    n, k = map(int,input().split(" "))
    arr = [int(input()) for i in range(n)]
    count = 0
    for i in range(1,len(arr)+1):
        if k // arr[-i] > 0:
            count += k // arr[-i]
            k = k % arr[-i]
    return count

print(solution())